wasm-chain-integration/src/v1/types.rs

```diff
@@ -773,8 +797,12 @@ impl<'a, BackingStore: trie::BackingStoreLoad> InstanceState<'a, BackingStore> {
     }
 
     /// Delete an entry. Return
-    /// - 0 if the entry was invalidated/deleted already
-    /// - 1 if an entry existed and was not invalidated.
+    /// - 1 if an entry was deleted
+    /// - u32::MAX if either
+    ///     - The entry did not exist or was invalidated/deleted already.
+    ///     - The generation of the entry provided did not match the current
+    ///       one.
+    ///     - The entry was part of a locked subtree.
```

Comment:
I suppose we could consider revising the API so it returns different codes for locked/non-existent.

Reply:
I suppose it could prove to be somewhat useful to the consumer being able to distinguish from these two cases.

Reply:
Yes, I agree.

To be consistent with `delete_prefix` I think the return codes should be 
```
    /// - 0 if the tree was locked
    /// - 1 if the entry did not exist or was already invalidated
    /// - 2 if something was deleted.
```